dcicutils/glacier_utils.py

```diff
@@ -0,0 +1,455 @@
+            PRINT(f'Object {bucket}/{key} copied back to its original location in S3')
+            return response
+        except Exception as e:
+            PRINT(f'Error copying object {bucket}/{key} back to its original location in S3: {str(e)}')
+            return None
+
+    def restore_glacier_phase_one_restore(self, atid_list: List[Union[dict, str]],
+                                          days: int = 7) -> (List[str], List[str]):
+        """ Triggers a restore operation for all @id in the @id list, returning a list of success and
+            error objects.
+
+        :param atid_list: list of @ids or actual file object metadata to restore from glacier
+        :param days: days to store the temporary copy
+        :return: 2 tuple of success, error list of @ids
+        """
+        success, errors = [], []
+        for atid in atid_list:
+            if isinstance(atid, dict):
+                _atid = atid['@id']
+                current_success, current_error = self.restore_portal_from_glacier(_atid, file_meta=atid, days=days)
+            else:
+                _atid = atid
+                success, current_error = self.restore_portal_from_glacier(atid, days=days)
+            if current_error:
+                PRINT(f'Failed to restore bucket/keys: {current_error}')
+                errors.append(_atid)
+            else:  # no errors should occur
+                success.append(_atid)
+        if len(errors) != 0:
+            PRINT(f'Errors encountered restoring @ids: {errors}')
+        else:
+            PRINT(f'Successfully triggered restore requests for all @ids passed {success}')
+        return success, errors
+
+    def restore_glacier_phase_two_copy(self, atid_list: List[Union[str, dict]], storage_class: str = 'STANDARD',
+                                       parallel: bool = False, num_threads: int = 4) -> (List[str], List[str]):
+        """ Triggers a copy operation for all restored objects passed in @id list
+
+        :param atid_list: list of @ids or actual file metadata objects to issue copy operations back to their
+                          original bucket/key location
+        :param storage_class: which storage class to copy into
+        :param parallel: whether to parallelize the copy
+        :param num_threads: number of threads to use when parallelizing, default to 4
+        :return: 2 tuple of success, error list of @ids
+        """
+        success, errors = [], []
+        if parallel:
+            with ThreadPoolExecutor(max_workers=num_threads) as executor:
+                futures = []
+                for atid in atid_list:
+                    if isinstance(atid, dict):
+                        _atid = atid['@id']
+                        files_meta = self.resolve_bucket_key_from_portal(_atid, atid)
+                    else:
+                        _atid = atid
+                        files_meta = self.resolve_bucket_key_from_portal(_atid)
```

Comment:
This general pattern of identifying the `@id` is repeated within this method and elsewhere and probably worthy of it's own helper method. May be worth it to reconsider the strategy of where the resolution of the string identifier vs. properties occurs such that it can be centralized in one place or as few locations as possible so separate method calls are not required (as is occurring with `resolve_bucket_key_from_portal` here). Consider renaming `atid_list` as well to reflect the fact that the contents are either `@id`s or properties.